File: 18-8/DashboardActivity.java
package com.example.permissionex;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.net.Uri;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

public class DashboardActivity extends AppCompatActivity
{
    TextView textView;
    Button btn1;
    SharedPreferences sharedPreferences;
    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_dashboard);

        textView=findViewById(R.id.txt1);
        btn1=findViewById(R.id.btn1);
        sharedPreferences=getSharedPreferences("user_session", Context.MODE_PRIVATE);


        Toast.makeText(DashboardActivity.this, "Logged in as "+sharedPreferences.getString("e1","error"), Toast.LENGTH_SHORT).show();

        btn1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                    sharedPreferences.edit().clear().commit();

                    startActivity(new Intent(DashboardActivity.this,MainActivity.class));


            }
        });
    }

    @Override
    public boolean onCreateOptionsMenu(@NonNull Menu menu)
    {
        getMenuInflater().inflate(R.menu.mymenu,menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {

        switch (item.getItemId())
        {

            case R.id.i1:

                String num="9999999999";
                Intent i =new Intent(Intent.ACTION_CALL);
                i.setData(Uri.parse("tel: "+num));
                startActivity(i);


                break;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: 10-11/Project/Activity/DashboardActivity.kt
package com.example.e_greetingsheet.Activity

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import android.widget.Toast
import com.example.e_greetingsheet.R
import com.example.e_greetingsheet.databinding.ActivityDashboardBinding
import com.example.e_greetingsheet.databinding.ActivityLoginBinding

class DashboardActivity : AppCompatActivity()
{
    private lateinit var binding: ActivityDashboardBinding
    lateinit var sharedPreferences: SharedPreferences
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityDashboardBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        sharedPreferences=getSharedPreferences("User_Session", Context.MODE_PRIVATE)

        Toast.makeText(applicationContext,"Welcome "+sharedPreferences.getString("mob",null),Toast.LENGTH_LONG).show()

    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {

        menuInflater.inflate(R.menu.option,menu)

        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {

        when(item.itemId)
        {
            R.id.logout->
            {
                sharedPreferences.edit().clear().commit()
                //finish()
                startActivity(Intent(this,Log::class.java))
            }
        }

        return super.onOptionsItemSelected(item)
    }
}